runconfig/hostconfig_unix.go

```diff
@@ -105,3 +106,55 @@ func ValidateQoS(hc *container.HostConfig) error {
 	}
 	return nil
 }
+
+// ValidateResources performs platform specific validation of the resource settings
+// cpu-rt-runtime and cpu-rt-period can not be greater than their parent, cpu-rt-runtime requires sys_nice
+func ValidateResources(hc *container.HostConfig, si *sysinfo.SysInfo) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
+
+	if hc.Resources.CPURtPeriod > 0 && !si.CPURtPeriod {
+		return fmt.Errorf("invalid --cpu-rt-period: Your kernel does not support cgroup rt period")
+	}
+
+	if hc.Resources.CPURtRuntime > 0 && !si.CPURtRuntime {
+		return fmt.Errorf("invalid --cpu-rt-runtime: Your kernel does not support cgroup rt runtime")
+	}
+
+	if hc.Resources.CPURtPeriod != 0 && hc.Resources.CPURtRuntime != 0 && hc.Resources.CPURtRuntime > hc.Resources.CPURtPeriod {
+		return fmt.Errorf("invalid --cpu-rt-runtime: rt runtime cannot be higher than rt period")
+	}
+
+	if hc.Resources.CPURtRuntime > 0 && !hasCap(hc, "SYS_NICE") {
+		return fmt.Errorf("invalid --cpu-rt-runtime: changing to rt priority requires sys_nice capability")
```

Comment:
I got overly ambition trying to save people from themselves. I'll get this removed.